src/components/AuthForm.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { useEffect, useState } from "react";
+import { object, string } from "yup";
+import classes from "../styles/AuthForm.module.scss";
+import { FieldNames } from "../common/enums/FieldNames";
+import { ERROR_MESSAGES } from "../common/constants/constants";
+import { useAuthentication } from "../common/hooks/AuthHooks";
+import { useLocation, useNavigate } from "react-router-dom";
+import Button from "./Button";
+
+const PASSWORD_MIN_CHAR = 6;
+
+interface AuthData {
+  username?: string;
+  email: string;
+  password: string;
+  confirmPassword?: string;
+}
+
+type FormError = {
+  fieldName: keyof AuthData;
+  message: string;
+};
+
+const AuthForm: React.FC = () => {
+  const [authData, setAuthData] = useState<AuthData>({
+    email: "",
+    password: "",
+  });
+
+  const [errors, setErrors] = useState<FormError[]>([]);
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const [isSuccess, setIsSuccess] = useState(false);
+
+  const { signIn, signUp } = useAuthentication();
+
+  const location = useLocation();
+
+  let navigate = useNavigate();
```

Comment:
it is ok to remove spacings between `use` hooks (:

Reply:
Any reason for navigate not being `const` but `let`? lets talk about this on mentorship.